tests/lib/formatters/html.js

```diff
@@ -0,0 +1,291 @@
+/**
+ * @fileoverview Tests for HTML reporter.
+ * @author Julian Laval
+ * @copyright 2015 Julian Laval. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var assert = require("chai").assert,
+    formatter = require("../../../lib/formatters/html");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Remove everything that isn't content
+ * @param {string} html Original content to be parsed
+ * @returns {string} The parsed content
+ */
+function parseHTML(html) {
+    // Match everything between <div id="overview"> and </table>, minus the date generated & the script
+    return html.match(/<d.*e>/)[0].replace(/ -[^<]*/, "");
+}
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+describe("formatter:html", function() {
+    describe("when passed a single error message", function() {
+
+        var code = [{
+            filePath: "foo.js",
+            errorCount: 1,
+            warningCount: 0,
+            messages: [{
+                message: "Unexpected foo.",
+                severity: 2,
+                line: 5,
+                column: 10,
+                ruleId: "foo",
+                source: "foo"
+            }]
+        }];
+
+        it("should return a string in HTML format with 1 issue in 1 file and styled accordingly", function() {
+            var result = formatter(code);
+            assert.equal(parseHTML(result), "<div id=\"overview\" class=\"bg-2\"><h1>ESLint Report</h1><div>1 problem (1 error, 0 warnings)</div></div><table><tbody><tr class=\"bg-2\" data-group=\"f-0\"><th colspan=\"4\">[+] foo.js <span>1 problem (1 error, 0 warnings)</span></th></tr><tr class=\"f-0\" style=\"display:none\"><td>5:10</td><td class=\"clr-2\">Error</td><td>Unexpected foo.</td><td><a href=\"http://eslint.org/docs/rules/foo\" target=\"_blank\">foo</a></td></tr></tbody></table>");
```

Comment:
We probably don't want to be testing exact HTML string output as that will be hard to maintain. Maybe use Cheerio to look for specific things in the DOM?